real_time/arduino_src/radio_buggy_mega/main.cpp

```diff
@@ -347,59 +344,32 @@ int main(void)
                         break;
                     default:
                         // report unknown message
-                        g_rbsm.Send(RBSM_MID_ERROR, RBSM_EID_RBSM_INVALID_MID);
                         dbg_printf("Got message with invalid mid %d and data %d\n",
                              new_command.message_id,
                              new_command.data);
+                        g_errors |= _BV(RBSM_EID_RBSM_INVALID_MID);
                         break;
                 } //End switch(new_command.message_id)
-            } 
+            }
+            // report stream losses for tracking
             else if(read_status == RBSM_ERROR_INVALID_MESSAGE) 
             {
-                // report stream losses for tracking
-                g_rbsm.Send(RBSM_MID_ERROR, RBSM_EID_RBSM_LOST_STREAM);
+                dbg_printf("RBSM could not parse message.\n");
+                g_errors |= _BV(RBSM_EID_RBSM_LOST_STREAM);
+            }
+            // should not be other faults
+            else {
+                dbg_printf("Unknown RBSM parse error.\n");
+                g_errors |= _BV(RBSM_EID_RBSM_LOST_STREAM);
```

Comment:
Do we want to set the same error bit for an unknown error condition?  It might make things harder to debug in the high level logs.